tasks/main.yml

```diff
@@ -0,0 +1,77 @@
+---
+- name: Update repositories cache (apt)
+  apt:
+    update_cache: yes
+  changed_when: false
+  when: ansible_pkg_mgr == "apt"
+
+- name: Install required packages (apt)
+  apt:
+    name: clamav
+  when: ansible_pkg_mgr == "apt"
+
+- name: Update repositories cache (pacman)
+  pacman:
+    update_cache: yes
+  changed_when: false
+  when: ansible_pkg_mgr == "pacman"
+
+- name: Install required packages (pacman)
+  pacman:
+    name: clamav
+  when: ansible_pkg_mgr == "pacman"
+
+- name: Create configurations
+  template:
+    src: "{{ item }}.conf.j2"
+    dest: /etc/clamav/{{ item }}.conf
+    owner: root
+    group: clamav
+    mode: 0640
+  notify: Restart clamav
+  with_items:
+    - clamd
+    - freshclam
+
+- name: Create units
+  template:
+    src: "{{ item }}.service.j2"
+    dest: /etc/systemd/system/{{ item }}.service
+    owner: root
+    group: root
+    mode: 0644
+  register: clamd_unit
+  with_items:
+    - clamd
+    - freshclamd
+
+- name: Create tmpfiles configuration
+  template:
+    src: tmpfiles.conf.j2
+    dest: /etc/tmpfiles.d/clamav.conf
+    owner: root
+    group: root
+    mode: 0644
+  register: tmpfiles
+
+- name: Create tmpfiles
+  command: systemd-tmpfiles --create
+  when: tmpfiles.changed
+
+- name: Reload systemd
+  command: systemctl daemon-reload
+  when: clamd_unit.changed
+
+- name: Download initial clam databases
+  command:
+    cmd: freshclamd
+    creates: /var/lib/clamav/main.cvd
+
+- name: Start services
```

Comment:
Is this necessary when you already have a handler for that?

Reply:
Yes

Reply:
How so?

Reply:
The config may not be changed but clam may not be running

Reply:
If you add `enabled: true` to the handler we don`t need the start task.
When the config is copied the first time, it most definitely is changed.

Reply:
But then the handler is changed when the unit was disabled and not running.

Reply:
Your task would also be changed?

Reply:
Okay. Proper use case: No configuration was changed, but clamd was not running, probably because it failed

Reply:
Okay. Proper use case: No configuration was changed, but clamd was not running, probably because it failed

Reply:
Is clamd prone to fail?
Wouldn`t proper monitoring or general automatic restarting of services be a better solution? :P

Reply:
Bruh.

Reply:
Maybe a dumb admin stopped it because it was causing too much CPU utilitation

Reply:
I don't think we will ever finish this project if we have to account for dumb admins.

Reply:
We'll not finish it either if you don't merge this